### PR TITLE
Fixes #29585: Hostname is added to In-Reply-To header only if needed

### PR DESCRIPTION
**Pull request by penguin86**
Hello, this is a proposal of fix for issue #29585 (https://mantisbt.org/bugs/view.php?id=29585).
Regards

### CODE REVIEW

File: core/email_api.php [hunk 1]
```diff
@@ -1405,7 +1405,11 @@ function email_send( EmailData $p_email_data ) {
 					$t_mail->set( 'MessageID', '<' . $t_value . '>' );
 					break;
 				case 'In-Reply-To':
-					$t_mail->addCustomHeader( $t_key . ': <' . $t_value . '@' . $t_mail->Hostname . '>' );
+					if( preg_match('/<.+@.+>/m', $t_value ) ) {
+						$t_mail->addCustomHeader( $t_key . ': ' . $t_value );
+					} else {
+						$t_mail->addCustomHeader( $t_key . ': <' . $t_value . '@' . $t_mail->Hostname . '>' );
+					}
```
[the remark this answers is not in the document]
Comment (penguin86) [reply core/email_api.php:1412]: I originally wrote the code like this, but I tought it was less clear. But sounds good to me, thanks!

**Comment by penguin86** [comment]
Hello @dregad, I made the proposed edits.
Ths strtolower() call was placed in every switch branch to keep the canonical header case, it seems more clear to me.
Thank you for your review.

**Comment by dregad** [comment]
> keep the canonical header case, it seems more clear to me

TBH I don't really see the added value from doing this... Makes code harder to read and adds unnecessary (even if negligible) overhead of 2 strtolower() calls per iteration.

**Comment by penguin86** [comment]
Hello @dregad, I made the proposed edit.
Thank you

**Review by dregad — APPROVED** [review]

**Comment by dregad** [comment]
Thanks. Waiting a few days before merge, to give other devs a chance to review.

**Comment by dregad** [comment]
Thanks for your contribution !

**Pull request merged**